hamilton/plugins/h_ddog.py

```diff
@@ -130,7 +106,10 @@ def run_before_graph_execution(self, *, run_id: str, **future_kwargs: Any):
         :param run_id: ID of the run
         :param future_kwargs: reserved for future keyword arguments/backwards compatibility.
         """
-        span = tracer.start_span(name=self.root_name, activate=True, service=self.service)
+        current_context = tracer.current_trace_context()
```

Comment:
This returns `None` if there's no active trace, so will be a safe no-op if there isn't one. `child_of` in `start_span` is `None` by default, so we're just assigning the parent if it exists.

Reply:
Good idea — added!